engine/extensions-mondrian/source/org/pentaho/reporting/engine/classic/extensions/datasources/mondrian/MondrianUtil.java

```diff
@@ -66,8 +67,11 @@ public static String parseSchemaName( final ResourceManager resourceManager,
     return null;
   }
 
-  private static String parseXmlDocument( final InputStream stream ) throws ResourceCreationException {
+  private static String parseXmlDocument( final InputStream stream ) throws ResourceCreationException,
+      ParserConfigurationException {
     final DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+    dbf.setFeature( XMLConstants.FEATURE_SECURE_PROCESSING, true );
```

Comment:
This is in the same .jar file as DocumentHelper above and has the same access to XMLParserFactoryProducer as DocumentHelper above. So please have this call call into XMLParserFactoryProducer the same as DocumentHelper (and everywhere else). Ultimately, there will be some other defect that we will need to fix, and fixing it in one place for PRD is better than fixing it in 6.